src/ensembl/src/content/app/browser/drawer/drawerActions.ts

```diff
@@ -1,16 +1,61 @@
 import { createAction } from 'typesafe-actions';
 
 import { getDrawerAnalyticsObject } from 'src/analyticsHelper';
+import { ActionCreator, Action, Dispatch } from 'redux';
+import { ThunkAction } from 'redux-thunk';
+import { RootState } from 'src/store';
+import { getBrowserActiveGenomeId } from '../browserSelectors';
 
-export const changeDrawerView = createAction(
-  'drawer/change-drawer-view',
+export const changeDrawerViewForGenome = createAction(
+  'drawer/update-drawer-view',
   (resolve) => {
-    return (drawerView: string) =>
+    return (drawerView: { [genomeId: string]: string }) =>
       resolve(drawerView, getDrawerAnalyticsObject('User Interaction'));
   }
 );
 
-export const toggleDrawer = createAction('drawer/toggle-drawer', (resolve) => {
-  return (drawerOpened?: boolean) =>
-    resolve(drawerOpened, getDrawerAnalyticsObject('User Interaction'));
-});
+export const changeDrawerView: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = (drawerViewForGenome: string) => (
+  dispatch: Dispatch,
```

Comment:
the dispatch in the thunk is not of type `Dispatch`, but of type `ThunkDispatch`, or something like that. Functions of type `Dispatch` can not dispatch other thunks; while functions of type `ThunkDispatch` can. In fact, if you delete `Dispatch`, you will see that typescript can infer the correct type of `dispatch` by itself, from the `ThunkAction` type.